### PR TITLE
Pipe/Validator related refactoring and code-cleanup

### PR DESCRIPTION
**Pull request by tnleeuw**

### CODE REVIEW

File: core/src/main/java/org/frankframework/core/IPipe.java [hunk 1]
```diff
@@ -186,4 +186,16 @@ default void throwEvent(String event) {
 	void throwEvent(String event, Message eventMessage);
 
 	boolean sizeStatisticsEnabled();
+
+	/**
+	 * Regular expression to mask strings in the log. For example, the regular expression <code>(?&lt;=&lt;password&gt;).*?(?=&lt;/password&gt;)</code>
+	 * will replace every character between keys '&lt;password&gt;' and '&lt;/password&gt;'. <b>note:</b> this feature is used at adapter level,
+	 * so one pipe affects all pipes in the pipeline (and multiple values in different pipes are merged)
```
[the remark this answers is not in the document]
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/core/IPipe.java:193]: It's copy-paste of existing JavaDoc from AbstractPipe so I did not review it for clarity... I'll rephrase that a bit.

Regarding full or partial match: should be partial match as it matches against log lines. Again, I'll double-check that code and clarify in the JavaDoc!

File: core/src/main/java/org/frankframework/pipes/XmlValidator.java [hunk 2]
```diff
@@ -428,67 +422,22 @@ public IValidator getResponseValidator() {
...
 	 * It overrides the stop and start method to prevent the original validator from being started and stopped.
 	 */
-	public static class ResponseValidatorWrapper implements IXmlValidator {
-
-		private @Getter @Setter String name;
-
-		private final Map<String, PipeForward> forwards = new HashMap<>();
-
-		protected XmlValidator owner;
+	public static class ResponseValidatorWrapper extends AbstractResponseValidatorWrapper<XmlValidator> implements IXmlValidator {
```
[the remark this answers is not in the document]
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/pipes/XmlValidator.java:425]: There's also one in the BaseResponseValidator. They were duplicating all the delegating-logic.

File: core/src/main/java/org/frankframework/core/IPipe.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/core/IPipe.java]: I hope that the updated JavaDoc covers what was unclear to you.

File: core/src/main/java/org/frankframework/pipes/XmlValidator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkosternl) [reply core/src/main/java/org/frankframework/pipes/XmlValidator.java:425]: Ah yes, I was looking at the right side of the diff. No I see it. Great.

File: core/src/main/java/org/frankframework/core/IPipe.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkosternl) [reply core/src/main/java/org/frankframework/core/IPipe.java]: Yes, much better. Thanks!

**Review by jkosternl — APPROVED** [review]

File: core/src/main/java/org/frankframework/core/IPipe.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/core/IPipe.java]: You're welcome!

File: core/src/main/java/org/frankframework/core/Adapter.java [hunk 3]
```diff
@@ -194,32 +193,26 @@ public void configure() throws ConfigurationException { //TODO check if we shoul
...
 
+	@Nonnull
+	String computeCombinedHideRegex() {
+		String combinedHideRegex = pipeline.getPipes().stream()
+				.map(IPipe::getHideRegex)
+				.filter(StringUtils::isNotEmpty)
+				.distinct()
+				.collect(Collectors.joining(")|(", "(", ")"));
+
+		if ("()".equals(combinedHideRegex)) return "";
```
Comment (evandongen) [inline core/src/main/java/org/frankframework/core/Adapter.java:212]: Kwestie van stijl denk ik - maar ik zou dit voor de leesbaarheid uitschrijven:

```suggestion
		if ("()".equals(combinedHideRegex)) {
			return "";
		}
```

File: core/src/main/java/org/frankframework/processors/MonitoringPipeProcessor.java [hunk 4]
```diff
@@ -57,7 +55,7 @@ protected PipeRunResult processPipe(@Nonnull PipeLine pipeLine, @Nonnull IPipe p
 			summary.record(pipeDuration);
 
 			if (pipe.getDurationThreshold() >= 0 && pipeDuration > pipe.getDurationThreshold()) {
-				durationLog.info("Pipe [" + pipe.getName() + "] of [" + pipeLine.getOwner().getName() + "] duration [" + pipeDuration + "] ms exceeds max [" + pipe.getDurationThreshold() + "], message [" + message + "]");
+				durationLog.info("Pipe [{}] of [{}] duration [{}] ms exceeds max [{}], message [{}]", pipe::getName, ()->pipeLine.getOwner().getName(), ()->pipeDuration, pipe::getDurationThreshold, ()->message);
```
Comment (evandongen) [inline core/src/main/java/org/frankframework/processors/MonitoringPipeProcessor.java:58]:
```suggestion
				durationLog.info("Pipe [{}] of [{}] duration [{}] ms exceeds max [{}], message [{}]", pipe::getName, () -> pipeLine.getOwner().getName(), () -> pipeDuration, pipe::getDurationThreshold, () -> message);
```

**Review by evandongen — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/frankframework/core/IPipe.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkosternl) [reply core/src/main/java/org/frankframework/core/IPipe.java]: Ik fix het wel met de hand.

**Review by evandongen — APPROVED** [review]

**Pull request merged**